packages/dnb-eufemia/src/components/date-picker/hooks/useSubmittedCalendarDates.ts

```diff
@@ -0,0 +1,26 @@
+import { useCallback, useRef } from 'react'
+import { DatePickerDates } from './useDates'
+
+export type SubmittedCalendarDates = Pick<
+  DatePickerDates,
+  'startDate' | 'endDate'
+>
+
+export default function useSubmittedCalendarDates() {
+  const submittedCalendarDatesRef = useRef<SubmittedCalendarDates>({})
+
+  const setSubmittedCalendarDates = useCallback(
+    (dates: SubmittedCalendarDates) => {
+      submittedCalendarDatesRef.current = {
+        ...submittedCalendarDatesRef.current,
+        ...dates,
+      }
```

Comment:
```suggestion
      submittedCalendarDatesRef.current = dates
```

I don't think we need to merge them. I think we simply can overwrite the existing dates.

Reply:
Agreed 🙏